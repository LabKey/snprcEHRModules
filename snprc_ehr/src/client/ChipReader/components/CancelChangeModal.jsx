import React from 'react'
import { Modal, Button } from 'react-bootstrap'

export default class CancelChangeModal extends React.PureComponent {
    render() {
        return (
          <Modal
            dialogClassName="center-modal"
            onHide={ this.props.noClick }
            show={ this.props.show }
          >
            <Modal.Header closeButton>
              <Modal.Title>{this.props.title}</Modal.Title>
            </Modal.Header>

            <Modal.Body>
              <p>{this.props.message}</p>
            </Modal.Body>

            <Modal.Footer>
              <Button onClick={ this.props.yesClick }>Yes</Button>
              <Button bsStyle="primary" onClick={ this.props.noClick }>No</Button>
            </Modal.Footer>
          </Modal>
        )
    }
}
