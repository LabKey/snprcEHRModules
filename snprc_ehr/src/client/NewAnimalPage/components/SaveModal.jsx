import React from 'react'
import { Modal, Button } from 'react-bootstrap'
import SummaryPanel from './SummaryPanel'

export default class SaveModal extends React.PureComponent {
    state = {
        disabled: false
    }

    onExit = () => {
        this.setState(() => (
            { disabled: false }
        ))
    }

    onSaveClick = () => {
        this.setState(() => (
            { disabled: true }
        ), this.props.onSaveClick())
    };

    onCloseClick = () => {
        this.setState(() => (
            { disabled: true }
        ), this.props.onCloseClick())
    };

    render() {
        return (
          <Modal
            backdrop="static"
            dialogClassName="custom-modal"
            keyboard
            onExit={ this.onExit }
            onHide={ this.onCloseClick }
            show={ this.props.show }
          >
            <Modal.Header closeButton>
              <Modal.Title>Okay to Save?</Modal.Title>
            </Modal.Header>

            <Modal.Body>
              <SummaryPanel
                infoMessages={ [{ key: 1, value: 'Please review data before saving.' },
                            { key: 2, value: 'Hover cursor over fields for full text.' }] }
                newAnimalData={ this.props.newAnimalData }
              />
            </Modal.Body>

            <Modal.Footer>
              <Button disabled={ this.state.disabled } onClick={ this.onCloseClick }>Close</Button>
              <Button bsStyle="primary" disabled={ this.state.disabled } onClick={ this.onSaveClick }>Save changes</Button>
            </Modal.Footer>
          </Modal>
        )
    }
}
