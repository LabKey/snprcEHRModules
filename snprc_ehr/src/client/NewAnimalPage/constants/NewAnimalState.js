import moment from 'moment';

const NewAnimalState = () => {

    return  ({
        currentPanel: 1,
        selectedOption: undefined,
        newAnimalData: {
            selectedOption: undefined,
            id: undefined,
            birthDate: { date: moment() },
            bdStatus: undefined,
            acquisitionType: undefined,
            acqDate: { date: moment() },
            gender: undefined,
            sire: undefined,
            dam: undefined,
            species: undefined,
            colony: undefined,
            animalAccount: undefined,
            ownerInstitution: undefined,
            responsibleInstitution: undefined,
            room: undefined,
            cage: { value: undefined },
            diet: undefined,
            pedigree: undefined,
            iacuc: undefined
        },
        summaryData: [],
        speciesList: [],
        acquisitionTypeList: [],
        potentialDamList: [],
        potentialSireList: [],
        locationList: [],
        accountList: [],
        colonyList: [],
        dietList: [],
        institutionList: [],
        iacucList: [],
        pedigreeList: [],
        bdStatusList: [],
        isLoading: true,
        hasError: false,
        preventNext: true,
        saveOk: false,
        showSaveModal: false,
        showCancelModal: false,
        showSpeciesChangeModal: false,
        errorMessage: undefined
    });
}

export default NewAnimalState;
