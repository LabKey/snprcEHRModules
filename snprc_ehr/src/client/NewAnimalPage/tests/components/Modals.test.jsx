import React from 'react'
import { shallow } from 'enzyme'
import SaveModal from '../../components/SaveModal'
import CancelChangeModal from '../../components/CancelChangeModal'
import NewAnimalState from '../../constants/NewAnimalState'

jest.mock('moment', () => {
    const moment = jest.requireActual('moment')('2023-01-01T00:00:00.000Z');
    return () => moment.utc(0);
});

describe('Modal tests', () => {
    test('Should render the SaveModal', () => {
        const { newAnimalData } = new NewAnimalState()

        const wrapper = shallow(
          <SaveModal
            newAnimalData={ newAnimalData }
            onCloseClick={ () => { return false } }
            onSaveClick={ () => { return false } }
            show
          />
        )
        expect(wrapper).toMatchSnapshot()
    })

    test('Should render the CancelChangeModal', () => {
        const wrapper = shallow(
          <CancelChangeModal
            message="Are you sure you want to cancel?"
            noClick={ () => { return false } }
            show
            title="Cancel changes?"
            yesClick={ () => { return false } }
          />
        )
        expect(wrapper).toMatchSnapshot()
    })
})
