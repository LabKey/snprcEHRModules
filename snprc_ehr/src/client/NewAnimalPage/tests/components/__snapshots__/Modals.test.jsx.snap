// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests Should render the CancelChangeModal 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="center-modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Cancel changes?
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <p>
      Are you sure you want to cancel?
    </p>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      Yes
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      No
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`Modal tests Should render the SaveModal 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="custom-modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Okay to Save?
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <SummaryPanel
      infoMessages={
        Array [
          Object {
            "key": 1,
            "value": "Please review data before saving.",
          },
          Object {
            "key": 2,
            "value": "Hover cursor over fields for full text.",
          },
        ]
      }
      newAnimalData={
        Object {
          "acqDate": Object {
            "date": "1970-01-01T00:00:00.000Z",
          },
          "acquisitionType": undefined,
          "animalAccount": undefined,
          "birthDate": Object {
            "date": "1970-01-01T00:00:00.000Z",
          },
          "cage": Object {
            "value": undefined,
          },
          "colony": undefined,
          "dam": undefined,
          "diet": undefined,
          "gender": undefined,
          "iacuc": undefined,
          "id": undefined,
          "ownerInstitution": undefined,
          "pedigree": undefined,
          "responsibleInstitution": undefined,
          "room": undefined,
          "sire": undefined,
          "species": undefined,
        }
      }
    />
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      Close
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      Save changes
    </Button>
  </ModalFooter>
</Modal>
`;
